providers/implementations/keymgmt/dsa_kmgmt.c

```diff
@@ -422,10 +422,15 @@ static void *dsa_gen_init(void *provctx, int selection)
 static int dsa_gen_set_template(void *genctx, void *templ)
 {
     struct dsa_gen_ctx *gctx = genctx;
+    EVP_PKEY *pkey = templ;
+    DSA *dsa;
```

Comment:
You're doing this because #10289 calls set_template incorrectly.  This is what you should do:

``` C
    DSA *dsa = templ;
```

Reply:
All you really need to do is undo the changes done on this file in d0aa5d591e08f7a63cfa285251c0dae311292a20

Reply:
ta - I am knee deep fixing up all the ffc parameter stuff at the moment..